react-backend/client/src/cb-admin/pages/Signup.js

```diff
@@ -68,26 +68,27 @@ export default class CbAdminSignup extends React.Component {
 
   submitHandler = (e) => {
     e.preventDefault();
-    console.log('Submit!');
-    // CbAdmin.create(payloadFromState(this.state))
-    //   .then(() => {
-    //     this.props.history.push('/cb/login?ref=signup');
-    //   })
-    //   .catch((error) => {
-    //     if (errorStatusEquals(401)) {
-    //       this.setState({ errors: getValidationErrors(error) });
-
-    //     } else if (errorStatusEquals(500)) {
-    //       this.history.push('/error/500');
-
-    //     } else if (errorStatusEquals(404)) {
-    //       this.history.push('/error/404');
-
-    //     } else {
-    //       this.history.push('/error/unknown');
-
-    //     }
-    //   });
+
+    CbAdmin.create(payloadFromState(this.state))
+      .then(() => {
+        this.props.history.push('/cb/login?ref=signup');
+      })
+      .catch((error) => {
```

Comment:
Wanna use my weird promise hack to avoid accessing `this.history` after the component has unmounted?

Reply:
It would be helpful to post a link so I don't have to go hunting around for this code:

https://github.com/TwinePlatform/DataPower/blob/077f7e5a18de50ba93fe8a9e6b0c5c1ef3f8ecef/react-backend/client/src/components/hoc/admin_login_check.js

Reply:
In any case, I think I'm probably gonna cover this in a separate HOC